source/js/openstreetmap/addMarkers.ts

```diff
@@ -0,0 +1,144 @@
+                        } else {
+                            this.map?.setView(latlng, 16);
+                        }
+                    }
+                    pushCoordinatesToBrowserHistory({ lat: location.lat, lng: location.lng });
+                });
+                this.markers?.addLayer(marker);
+                if (location.element) {
+                    this.markerElementPairs?.push({marker: marker, element: location.element});
+                }
+            }
+        });
+        this.markers?.addTo(this.map as LeafletMap);
+    }
+
+    private createMarker(customIcon: Icon | undefined) {
+        let template = this.container?.querySelector('.c-openstreetmap__pin-icon');
+        let html = template?.innerHTML;
+        let icon = customIcon?.icon ? customIcon.icon : 'location_on';
+        let color = customIcon?.backgroundColor
+            ? customIcon.backgroundColor
+            : this.getPrimaryColor();
+        
+            if (!html) return;
+        html = html
+            .replace('{icon-name}', icon as string)
+            .replaceAll('{ICON_NAME}', icon as string)
+            .replace('{ICON_BACKGROUND_COLOR}', color as string);
+        let marker = L.divIcon({
+            className: 'c-openstreetmap__icon',
+            html: html,
+        });
+
+        return marker;
+    }
+
+    private createTooltip(tooltip: Tooltip) {
+        let template = this.container?.querySelector('.c-openstreetmap__pin-tooltip');
+        let clone = template?.cloneNode(true) as HTMLTemplateElement;
+
+        if (!tooltip.image?.src) {
+            const figure = clone.content.querySelector('figure');
+            if (figure) {
+                figure.remove();
+            }
+        }
+
+        if (!tooltip.url) {
+            let link = clone.content.querySelector('.c-openstreetmap__tooltip-link');
+            let title = clone.content.querySelector('.c-openstreetmap__tooltip-title');
+            const parent = link?.parentNode;
+
+            if (parent && title && link) {
+                parent.insertBefore(title, link);
+                link.remove();
+            }
+        }
+        let html = clone.innerHTML;
+        html = html
+            .replace('{TOOLTIP_HEADING}', tooltip.title ? tooltip.title as string : '')
+            .replace('{TOOLTIP_DIRECTIONS_URL}', tooltip.directions?.url ? tooltip.directions.url as string : '')
+            .replace('{TOOLTIP_DIRECTIONS_LABEL}', tooltip.directions?.label ? tooltip.directions.label as string : '')
+            .replace('{TOOLTIP_EXCERPT}', tooltip.excerpt ? tooltip.excerpt as string : '')
+            .replace('{TOOLTIP_IMAGE_SRC}', tooltip.image?.src ? tooltip.image.src as string : '')
+            .replace('{TOOLTIP_IMAGE_ALT}', tooltip.image?.alt ? tooltip.image.alt as string : '')
+            .replace('{TOOLTIP_LINK}', tooltip.url ? tooltip.url as string : '');
+        return html;
+    }
+
+    public markerElementObjects() {
```

Comment:
```suggestion
    public getMarkerElementObjects() {
```

Reply:
This is fixed, since it collided with naming the variable name :)